### PR TITLE
Upgrade CoreDNS to v1.11.1

### PR DESCRIPTION
**Pull request by mcharriere**
Signed-off-by: Matias Charriere <matias@giantswarm.io>

Towards https://github.com/giantswarm/roadmap/issues/2800

### CODE REVIEW

File: helm/coredns-app/templates/deployment-masters.yaml [hunk 1]
```diff
@@ -73,6 +73,8 @@ spec:
         securityContext:
           allowPrivilegeEscalation: false
           capabilities:
+            add:
+            - NET_BIND_SERVICE
```
Comment (mcharriere) [inline helm/coredns-app/templates/deployment-masters.yaml:77]: This is required as the image now sets this cap (https://github.com/coredns/coredns/pull/5969/files#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557R13)

**Review by Gacko — APPROVED** [review]

File: helm/coredns-app/templates/deployment-masters.yaml (hunk 1 above)
Comment (ubergesundheit) [reply helm/coredns-app/templates/deployment-masters.yaml:77]: Does this need some kyverno exception in regards to PSS? I faintly remember this isn't a problem in kube-system, right?
I also remember there was a mechanism for restricting the installation namespace of a chart/app. If above is true, the chart should have something like this in place (if not already)

Comment (Gacko) [reply helm/coredns-app/templates/deployment-masters.yaml:77]: Actually we are using this as a default in Ingress NGINX Controller for a while now. 😅

Comment (mcharriere) [reply helm/coredns-app/templates/deployment-masters.yaml:77]: Indeed, as Marco mentioned, it is allowed in PSS, even under restricted. 

> Containers must drop ALL capabilities, and are only permitted to add back the NET_BIND_SERVICE capability. [This is Linux only policy](https://kubernetes.io/docs/concepts/security/pod-security-standards/#os-specific-policy-controls) in v1.25+ (.spec.os.name != "windows")

You can find more info here: https://kubernetes.io/docs/concepts/security/pod-security-standards/#restricted

**Pull request merged**